4_Zadania_dodatkowe/6.InfiniteSlider/js/app.js

```diff
@@ -0,0 +1,71 @@
+$(document).ready(function () {
+   // console.log('test');
+
+    var elementPrevButton = $('#prevPicture');
+    var elementNextButton = $('#nextPicture');
+    // console.log(elementPrevButton, elementNextButton);
+
+    var elementListLi = $('li');
+    // console.log(elementListLi);
+
+    var elementUl = $('ul');
+    // console.log(elementUl);
+
+    var visibleImageIndex = 1;
+    var imageWidth = 404;
+    // console.log(visibleImageIndex, imageWidth);
+
+    var firstElement = elementListLi.first().clone();
+    var lastElement = elementListLi.last().clone();
+    // console.log(firstElement);
+    // console.log(lastElement);
+    elementUl.append(firstElement);
+    elementUl.prepend(lastElement);
+
+
+    function moveRight() {
+        visibleImageIndex++;
+        elementUl.animate({
+                'right': imageWidth * visibleImageIndex,
+            },
+            {
+                complete: function () {
+                    if (visibleImageIndex > 6) {
+                        visibleImageIndex = 1;
+                        elementUl.css('right', 404);
+                        // clearInterval(interval);
+                    }
+                }
+            });
+    }
+
+    function moveLeft() {
```

Comment:
Noticed `moveRight` and `moveLeft` functions are very similar? Whenever you find yourself writing the same code twice ask yourself seriously: can I express it once. Here it would be possible to create `move` function which accepts a parameter so `move(-1)` moves left and `move(1)` moves right. Code deduplication is a way to simplify code, avoid bugs, to make code more maintainable.

Reply:
Updated in the newest commit. Even though I included two if conditions in it so I'm not sure if it's the best solution